engine/lib/phx/src/engine/engine.rs

```diff
@@ -26,6 +35,12 @@ pub struct Engine {
     pub event_bus: EventBus,
     pub task_queue: TaskQueue,
     pub lua: Rf<Lua>,
+    /// Handle to the dedicated render thread (when multithreaded rendering is enabled)
+    pub render_thread: Option<Arc<RenderThreadHandle>>,
```

Comment:
I think we don't really need to use Arc. We can just use RenderThread directly. The same way we do with HmGui, EventBus and TaskQueue.